cumulusci/core/tests/test_config.py

```diff
@@ -210,20 +248,328 @@ def get_service(self, name):
+
+    def test_repo_url_from_repo_info(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        config._repo_info = {
+            'url': 'https://github.com/SalesforceFoundation/CumulusCI',
+        }
+        self.assertEqual(
+            'https://github.com/SalesforceFoundation/CumulusCI', config.repo_url)
+
+    def test_repo_url_no_repo_root(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        with temporary_dir() as d:
+            cwd = os.getcwd()
+            os.chdir(d)
+            self.assertIsNone(config.repo_url)
+            os.chdir(cwd)
+
+    def test_repo_url_from_git(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        self.assertIn('SalesforceFoundation/CumulusCI', config.repo_url)
+
+    def test_repo_owner_from_repo_info(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        config._repo_info = {
+            'owner': 'SalesforceFoundation',
+        }
+        self.assertEqual(
+            'SalesforceFoundation', config.repo_owner)
+
+    def test_repo_owner_no_repo_root(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        with temporary_dir() as d:
+            cwd = os.getcwd()
+            os.chdir(d)
+            self.assertIsNone(config.repo_owner)
+            os.chdir(cwd)
+
+    def test_repo_branch_from_repo_info(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        config._repo_info = {
+            'branch': 'master',
+        }
+        self.assertEqual(
+            'master', config.repo_branch)
+
+    def test_repo_branch_no_repo_root(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        with temporary_dir() as d:
+            cwd = os.getcwd()
+            os.chdir(d)
+            self.assertIsNone(config.repo_branch)
+            os.chdir(cwd)
+
+    def test_repo_commit_from_repo_info(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        config._repo_info = {
+            'commit': 'abcdef',
+        }
+        self.assertEqual(
+            'abcdef', config.repo_commit)
+
+    def test_repo_commit_no_repo_root(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        with temporary_dir() as d:
+            cwd = os.getcwd()
+            os.chdir(d)
+            self.assertIsNone(config.repo_commit)
+            os.chdir(cwd)
+
+    def test_repo_commit_no_repo_branch(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        with temporary_dir() as d:
+            cwd = os.getcwd()
+            os.chdir(d)
+            os.mkdir(os.path.join(d, '.git'))
+            with open(os.path.join(d, '.git', 'HEAD'), 'w') as f:
+                f.write('abcdef')
+
+            self.assertIsNone(config.repo_commit)
+
+            os.chdir(cwd)
+
+    def test_repo_commit_packed_refs(self):
+        config = BaseProjectConfig(BaseGlobalConfig())
+        with temporary_dir() as d:
+            cwd = os.getcwd()
```

Comment:
didn't we just add a cwd context manager? oh wait was that metadeploy?

Reply:
It was metadeploy. But it looks like most places that we're using `temporary_dir` in CumulusCI we do want to chdir to it, so I'll adjust it to handle that...